github/connection_config.go

```diff
@@ -5,8 +5,11 @@ import (
 )
 
 type githubConfig struct {
-	Token   *string `hcl:"token"`
-	BaseURL *string `hcl:"base_url"`
+	Token          *string `hcl:"token,optional"`
+	BaseURL        *string `hcl:"base_url,optional"`
+	AppId          *int64  `hcl:"app_id,optional"`
```

Comment:
Can you please rename the config args to:
- `app_id` (no change)
- `app_installation_id`
- `app_private_key`

This helps to group the config args together and protects us against conflicts if there are any other installation IDs/private keys in GitHub we may add in the future